flowpipe/node.py

```diff
@@ -160,6 +168,64 @@ def evaluate(self):
 
         return outputs
 
+    def _evaluate(self, **inputs):
+        # Look if iteration is needed
+        # Iterables is a dictionary with each entry an list of inputs to iterate over later-on
+        is_iteration_detected, iterables = False, {}
+        # If itertation is disabled just continue normally
+        if self.iteration_count != -1:
+            # Iterate over the inputs
+            for key, value in inputs.items():
+                # Input value is an IterationPlug so iteration is needed
+                if isinstance(value, IterationPlug):  
+                    is_iteration_detected = True
+                    iterables[key] = [{key: x} for x in value]  # Create a list of kwargs to iterate over lateron.
+                    if self.iteration_count is not None:
+                        # Check the number of iterations if provided
+                        assert len(value) == self.iteration_count, f"Length ({len(value)}) of iterable `{key}` is unequal to expected length {self.iteration_count}"
```

Comment:
This situation should raise a `ValueError`.

Reply:
Note that `assert` statements can get stripped from the code in some scenarios and are purely for development-time debugging. cf. https://docs.python.org/3/reference/simple_stmts.html#assert which states that the execution of `assert` depends on the `__debug__` flag, which will be set to `False` if optimized execution is requested.

Reply:
That's the main reason I use `assert` to be honest. Type checking is not very pythonic it only helps a lot with debugging. When the code is finished the `assert` statements are all ignored which improves calculation time (especially if you have a lot of them).

But you also have a valid argument 👍